core/variant/variant.cpp

```diff
@@ -1311,9 +1311,11 @@ void Variant::_clear_internal() {
 			_get_obj().obj = nullptr;
 			_get_obj().id = ObjectID();
 		} break;
-		case _RID: {
+		case RID: {
 			// not much need probably
-			reinterpret_cast<RID *>(_data._mem)->~RID();
+			// Can't seem to use destructor + scoping operator, so hack.
+			typedef ::RID RID_Class;
+			reinterpret_cast<RID_Class *>(_data._mem)->~RID_Class();
```

Comment:
I couldn't get either `~::RID()` or `::~RID()` to compile, it seems combining those operators doesn't work in C++ (unless someone knows a trick here, maybe with some parentheses?).

Anyway, the `typedef` approach works, and makes me think that we could typedef `AABB` and `RID` in the relevant files instead of using `::AABB` and `::RID` if we want to be explicit and allow easier grepping.